src/SDL_hints.c

```diff
@@ -160,37 +187,41 @@ SDL_bool SDL_SetHint(const char *name, const char *value)
 
 const char *SDL_GetHint(const char *name)
 {
-    const char *env;
-    SDL_Hint *hint;
-
     if (!name) {
         return NULL;
     }
 
-    env = SDL_getenv(name);
-    for (hint = SDL_hints; hint; hint = hint->next) {
-        if (SDL_strcmp(name, hint->name) == 0) {
-            if (!env || hint->priority == SDL_HINT_OVERRIDE) {
-                return hint->value;
-            }
-            break;
+    const SDL_PropertiesID hints = GetHintProperties();
+    if (!hints) {
+        return NULL;
+    }
+
+    const char *env = SDL_getenv(name);
+
+    SDL_LockProperties(hints);
+
+    SDL_Hint *hint = SDL_GetProperty(hints, name, NULL);
+    const char *retval = env;
+    if (hint) {
+        if (!env || hint->priority == SDL_HINT_OVERRIDE) {
+            retval = hint->value;
         }
     }
-    return env;
+
+    SDL_UnlockProperties(hints);
+
+    return retval;
 }
 
 int SDL_GetStringInteger(const char *value, int default_value)
 {
     if (!value || !*value) {
         return default_value;
-    }
-    if (*value == '0' || SDL_strcasecmp(value, "false") == 0) {
+    } else if (*value == '0' || SDL_strcasecmp(value, "false") == 0) {
```

Comment:
Why are you adding `else if` for short circuit return values?

Reply:
Packs the code in tighter, since you don't have a single `{` on a separate line, and it's logically a single block of early exit conditions.

But that's just me, it's not super-important if you prefer it the other way.

Reply:
I usually do, but I see your point about a block of early-out conditions. I think I'd slightly prefer to leave it the way it was for consistency with other code.

Reply:
Backed out these `else if` changes in those two functions (SDL_GetStringInteger and SDL_GetStringBoolean) from the first commit and force-pushed to 336cf87275a3dcec851b81a4e33d6b4d6a6693fb.